public/js/livesearch.js

```diff
@@ -0,0 +1,104 @@
+	//init vars
+	var input;
+	var results;
+	var indiResults;	
+	var urlResult;
+	var formMethod;
+	var typeData;
+	var resultFilter;
+	var identifier ;
+
+	//set values to vars, this is done wherever the livesearch is called
+	function setInputListener(inputField , resultField, resultHolder, uUrl, method, typeD){
+		
+		input = inputField;
+		results = resultField;
+		indiResults = resultHolder;
+		urlResult =  uUrl + '/';
+		typeData = typeD;
+		return this;
+	}	
+	
+	//set the method type for ajax submission
+	function setMethod(method){
+		var regex= /(GET|PUT)/g;
+		var methods = method.match(regex);
+		if(methods != null){
+			return this;
+		}
+		console.log("The method was not set correctly");
+		return false;
+	}
+
+	//setting the filter / find for returning only desired result html
+	function setResultFilter(filter){
+		resultFilter = filter;
+		return this;
+	}
+	
+	//debugging
+	//console.log("live search is loaded");
+
+	//actual jquery live search fn, takes in the form where live search is being performed
+	(function( $form ){
+		$.fn.liveSearch = function() {	
+			// debugging
+			// console.log($(this));
+			
+			//giving the form it's own var for simplicity sake, later functions redefine the 'this' method 
+			var uForm = $(this);
+			
+			//taking cleaning user supplied input - no extra paces or , will be queried
+			function filterResults(raw){
+				var filtered = raw.replace(/\s+/g,' ').trim().replace(',','');
+				return filtered;
+			}
+			//set the form input element text to cleaned text, nice and tidy
+			$(this).find(":text").val(filterResults($(this).find(":text").first().val()));
```

Comment:
You created the `uForm` variable to replace having type `$(this)`, should you use `uForm` here to replace the `$(this)`?